src/dex-ui/services/HederaService/HederaService.ts

```diff
@@ -76,139 +92,142 @@ function createHederaService() {
+    const addLiquidityTransaction = await new ContractExecuteTransaction()
       .setContractId(addLiquidityContractId)
       .setGas(9000000)
       .setFunction(
         "addLiquidity",
         new ContractFunctionParameters()
-          .addAddress(fromAddress)
+          .addAddress(walletAddress)
           .addAddress(firstTokenAddr)
           .addAddress(secondTokenAddr)
-          .addInt64(firstTokenQty)
-          .addInt64(secondTokenQty)
+          .addInt256(firstTokenQty)
+          .addInt256(secondTokenQty)
       )
-      .setNodeAccountIds([new AccountId(3)]);
-
-    if (signer) {
-      const addLiquidityTxWalletSigned = await addLiquidityTxParams.freezeWithSigner(signer);
-      const addLiquidityTxWalletSignedRes = await addLiquidityTxWalletSigned.executeWithSigner(signer);
-      console.log(addLiquidityTxWalletSignedRes);
-
-      // TODO: for some reason this gives the error: addLiquidityTxWalletSignedRes.getReceiptWithSigner
-      // is not a function
-      // const addLiquidityTxWalletSignedReceipt = await addLiquidityTxWalletSignedRes.getReceiptWithSigner(signer)
-      // console.log(`Liquidity added status: ${addLiquidityTxWalletSignedReceipt.status}`);
-    } else {
-      const addLiquidityTx = await addLiquidityTxParams.freezeWith(client).sign(treasureKey);
-      const addLiquidityTxRes = await addLiquidityTx.execute(client);
-      const transferTokenRx = await addLiquidityTxRes.getReceipt(client);
-      console.log(`Liquidity added status: ${transferTokenRx.status}`);
-    }
+      .setNodeAccountIds([new AccountId(3)])
+      .freezeWithSigner(signer);
 
-    await pairCurrentPosition();
+    const result = await addLiquidityTransaction.executeWithSigner(signer);
+    console.log(result);
   };
 
   const removeLiquidity = async () => {
-    const tokenAQty = new BigNumber(3);
-    const tokenBQty = new BigNumber(3);
+    const tokenAQty = withPrecision(3);
+    const tokenBQty = withPrecision(3);
     console.log(`Removing ${tokenAQty} units of token A and ${tokenBQty} units of token B from the pool.`);
     const removeLiquidity = await new ContractExecuteTransaction()
       .setContractId(contractId)
       .setGas(2000000)
       .setFunction(
         "removeLiquidity",
         new ContractFunctionParameters()
-          .addAddress(treasure)
+          .addAddress(treasuryId.toSolidityAddress())
           .addAddress(tokenA)
           .addAddress(tokenB)
           .addInt64(tokenAQty)
           .addInt64(tokenBQty)
       )
       .freezeWith(client)
-      .sign(treasureKey);
+      .sign(treasuryKey);
     const removeLiquidityTx = await removeLiquidity.execute(client);
     const transferTokenRx = await removeLiquidityTx.getReceipt(client);
 
     console.log(`Liquidity remove status: ${transferTokenRx.status}`);
-    await pairCurrentPosition();
+    await pairCurrentPosition(contractId);
+  };
+
+  const swapToken = async ({
```

Comment:
is this being used anywhere? looks like in the swap slice it directly calls the swapToken function on the contract instead of using this... if we were to consume this method to do the swap we want to use the signer instead of the client too

Reply:
No but we should be calling the swapToken function directly from the HederaService (`HederaService.swapToken`). All fetching and async calls should be handled at the service layer. So, the logic in the swapSlice swap transaction call should be moved to the HederaService. I left this as-is for now since the PR is already huge and the change is not directly related to this task.